discordbot/cogs/internet.py

```diff
@@ -39,30 +53,35 @@ def __init__(self, bot: DiscordBot):
             user = user.display_name
         else:
             user = ctx.message.author.display_name
-        await ctx.send("https://robohash.org/{}.png".format(user.replace(" ", "%20")))
+        k = "https://robohash.org/{}.png".format(user.replace(" ", "%20"))
+        o = wget.download(k)
+        await ctx.send(file=o)
 
     @commands.command()
-    async def xkcd(self, ctx, query=""):
-        """Queries a random XKCD comic.
-
-        Do `^xkcd <number from 1-1662>` to pick a specific comic."""
-        if not query:
-            i = randint(1, 1662)
-            url = "https://xkcd.com/{}/".format(i)
-        elif query.isdigit() and 1 <= int(query) <= 1796:
-            url = "https://xkcd.com/{}/".format(query)
-        elif int(query) <= 0 or int(query) >= 1796:
-            await ctx.send("It has to be between 1 and 1796!")
-        elif not query.isdigit():
-            await ctx.send("You have to put a number!")
-        else:
-            await ctx.send("I don't know how you managed to do it, but you broke it.")
-        with aiohttp.ClientSession() as session:
-            async with session.get(url) as resp:
-                r = await resp.read()
-        resp = BSoup(r, "html.parser")
-        await ctx.send(
-            ":mag:**" + resp("img")[1]["alt"] + "**\nhttp:" + resp("img")[1]["src"] + "\n" + resp("img")[1]["title"])
+    async def xkcd(ctx, num=None):
+        """Looks up a xkcd comic"""
+        a = requests.get('https://xkcd.com/{}/info.0.json'.format(num))
```

Comment:
In my opinion, aiohttp is not strong enough _yet_ to provide good synchronicity. That being said, it feels incredibly slower than requests and os. I just don't think its a value option.

And your bot popularity should. There's no point in being the fastest bot around if you don't have the userbase that wants it. wget looks like a better option to download files currently.